README.md

```diff
@@ -1,2 +1,74 @@
 # Saga [![Build Status](https://travis-ci.org/ServiceComb/saga.svg?branch=master)](https://travis-ci.org/ServiceComb/saga?branch=master) [![Coverage Status](https://coveralls.io/repos/github/ServiceComb/saga/badge.svg?branch=master)](https://coveralls.io/github/ServiceComb/saga?branch=master) [![License](https://img.shields.io/badge/license-Apache%202-4EB1BA.svg)](https://www.apache.org/licenses/LICENSE-2.0.html)
 
+## Purpose
+Saga is implemented based on the compensation transaction driven mechanism, which provides a simple way to help users solve the data consistency problems encountered in micro-service applications.
```

Comment:
Saga is a type of [Compensating Transaction pattern](https://docs.microsoft.com/en-us/azure/architecture/patterns/compensating-transaction)

Reply:
modified completed